packages/dev/core/src/Materials/Textures/cubeTexture.ts

```diff
@@ -337,6 +340,33 @@ export class CubeTexture extends BaseTexture {
         }
 
         this._textureMatrix = value;
+
+        if (!this.getScene()?.useRightHandedSystem) {
+            return;
+        }
+
+        const scale = TmpVectors.Vector3[0];
+        const quat = TmpVectors.Quaternion[0];
+        const trans = TmpVectors.Vector3[1];
+        const euler = TmpVectors.Vector3[2];
+
+        this._textureMatrix.decompose(scale, quat, trans);
+        quat.toEulerAnglesToRef(euler);
+
+        euler.x *= -1;
+        euler.y *= -1;
+
+        Quaternion.FromEulerAnglesToRef(euler.x, euler.y, euler.z, quat);

```

Comment:
You can do it straight in the quaternion, multiplying z and w * -1. Unless I missed a step somewhere :-)
Consider that resolved if direct change to the quat doesn't work